### PR TITLE
Feature/add backchannel capability

### PR DESCRIPTION
**Pull request by paddy-sr**
Added backchannel capability. 
Depends on 6.2.0

### CODE REVIEW

File: src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/RequestLogicHelper.kt [hunk 1]
```diff
@@ -98,48 +112,92 @@ class RequestLogicHelper(private val config: FrejaEidAuthenticatorPluginConfig)
         return emptyList()
     }
 
-    fun getAuthTransaction(postData: Map<String, Any>): Map<String, Any>
+    fun doAuthenticate(postData: Map<String, Any>): Map<String, Any>
```
Comment (travisspencer) [inline src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/RequestLogicHelper.kt]: please don't use `doXyz` as names. `authenticate` is a verb and makes for a great name

File: src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/RequestLogicHelper.kt [hunk 2]
```diff
@@ -98,48 +112,92 @@ class RequestLogicHelper(private val config: FrejaEidAuthenticatorPluginConfig)
...
+    }
 
-        if (statusCode != 200)
-        {
-            if (_logger.isWarnEnabled)
-            {
-                _logger.warn("Got error response from authentication endpoint: error = {}, {}", statusCode,
-                        httpResponse.body(HttpResponse.asString()))
-            }
+    fun doSign(postData: Map<String, Any>): Map<String, Any>
```
Comment (travisspencer) [inline src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/RequestLogicHelper.kt]: same comment here

**Review by travisspencer — CHANGES_REQUESTED** [review]

**Comment by duraisamysathya** [comment]
tested with all the flows. Works as expected

File: src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/FrejaConstants.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.curity.identityserver.plugin.frejaeid.authentication
+
+class FrejaConstants
+{
+    companion object
```
Comment (renatoathaydes) [inline src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/FrejaConstants.kt:21]: You can do `object FrejaConstants` and remove the inner object.

**Review by renatoathaydes — APPROVED** [review]

**Review by travisspencer — APPROVED** [review]

**Pull request merged**